server/handlers/clients.js

```diff
@@ -0,0 +1,256 @@
+const Config = require('bcfg');
+const assert = require('bsert');
+
+const { loadClientConfigs } = require('../loadConfigs');
+
+const { configHelpers, clientFactory } = require('../utils');
+const {
+  getDefaultConfig,
+  createClientConfig,
+  createConfigsMap,
+  testConfigOptions,
+  deleteConfig,
+  getConfig
+} = configHelpers;
+
+let configsMap = null;
```

Comment:
Could we create the configs map at boot time right here?

Reply:
So there would need to be some sort of filesystem watch, which we already have, maybe that is just built into building the configsMap, and it can take an option `{ watch: Bool }`

Reply:
I don't follow. The file system watch we have is what restarts the server (this is actually always run though: https://github.com/bpanel-org/bpanel/blob/development/server/index.js#L82). When the server is restarted, then this would presumably be re-created. 

I say this isn't ideal though because it's not great that your server essentially has to shut down every time a new config is added. This makes me think though that now that we have this clients + configs api in place, we might not need to restart the server anymore and in which case it might actually be better to retrieve this map on every request (would be better if it was memoized in some way though so that it only re-calculates when there's been an update)

Reply:
What I meant was that a fs watch could rebuild the global configsMap on change instead of restarting the server

Reply:
You could turn off the watching if you wanted to at boot time. Any files written by `POST config` would be automatically added because the fs watch would see the change and rebuild the map

Reply:
Ah, i found one of the reasons I did it this way (not at boot time)- you need the config from the `req` object.